### DISCUSSION TITLE
Versioning paradigm

### DISCUSSION THREAD

**Discussion by TamirFriedman-RecoLabs** [shadcn-ui/ui#1467] [Q&A]
as Shadcn is not a library, I ask how would you (and us) handle versioning? does it will be versioning per component instead of global versioning? for example, if I added a button component before shad supported themes, and added a dropdown after, then one support theming and the other not? should every copy-pasted component should have its cookiecutted version at on top comment? how do we handle updates of components? when new features arrive, or even if there are security updates (imagine a security issue with the forms). as you see, it raises a lot of questions regarding versioning, updating, and generally tracking the code as it changes by both sides (the user programer, and the "library" provider)

**Comment by elpddev** [comment]
Related to: https://github.com/shadcn-ui/ui/issues/191
